src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1894,13 +1894,43 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RequestSEDActiv
             mActiveModeConsumers--;
     }
 
+    if (!onOff && delayIdle && CHIP_DEVICE_CONFIG_SED_ACTIVE_THRESHOLD.count() != 0)
+    {
+        err = DeviceLayer::SystemLayer().StartTimer(CHIP_DEVICE_CONFIG_SED_ACTIVE_THRESHOLD, RequestSEDModeChange, this);
+        if (CHIP_NO_ERROR == err)
+        {
+            return err;
+        }
+
+        ChipLogError(DeviceLayer, "Failed to postponed Idle Mode with error %" CHIP_ERROR_FORMAT, err.Format());
+    }
+
+    return SEDChangeMode();
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::SEDChangeMode()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    ConnectivityManager::SEDIntervalMode mode;
+
     mode = mActiveModeConsumers > 0 ? ConnectivityManager::SEDIntervalMode::Active : ConnectivityManager::SEDIntervalMode::Idle;
 
     if (mIntervalsMode != mode)
         err = SetSEDIntervalMode(mode);
 
     return err;
 }
+
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::RequestSEDModeChange(chip::System::Layer * apSystemLayer,
+                                                                               void * apAppState)
+{
+    if (apAppState != nullptr)
+    {
+        static_cast<GenericThreadStackManagerImpl_OpenThread *>(apAppState)->SEDChangeMode();
```

Comment:
This looks wrong to me.

Let's say we go into active mode for an exchange, and then the exchange completes.  We will start a timer to call this function.

If, before the timer fires, we enter active mode for some reason other than an exchange and then exit active mode, we will exit _immediately_, without waiting for this timer to fire.

What I think we need to do is:

1. If we fail to start the timer, should we perhaps update immediately instead of staying in active mode indefinitely?
2. The timer starting successfully, if there was no timer so far, should count as an active mode consumer (and increase mActiveModeConsumers by 1).  When the timer fires it should then decrement mActiveModeConsumers by 1.

Or something else to deal with this problem.

Reply:
@bzbarsky-apple It's not a problem as per the spec. section 2.12.5 specify SHOULD stay active, the case you've described is a possible corner case that doesn't go against this implementation. However this would have to be address when the Client Monitoring cluster will fully be implemented.

Reply:
This behavior is _allowed_ by the spec, but it's still not the intended behavior of this implementation as far as I can tell...